### ISSUE TITLE
RPA.WINDOWS Click not successful but no error reported again

### ISSUE THREAD

**Issue by hhsem** [robocorp/rpaframework#913]
Ref to 901:
I have installed rpaframework 22.5.0 and repeated the robot but still i got the error

 
Start / End / Elapsed:	20230405 17:25:21.170 / 20230405 17:25:30.669 / 00:00:09.499
00:00:09.498ITERATION
Start / End / Elapsed:	20230405 17:25:21.170 / 20230405 17:25:30.668 / 00:00:09.498
00:00:09.491TRY
Start / End / Elapsed:	20230405 17:25:21.171 / 20230405 17:25:30.662 / 00:00:09.491
00:00:00.521KEYWORD RPA.Windows . Control Window ${Window}
00:00:00.005KEYWORD RPA.Windows . Click ${Group}
Documentation:	
Mouse click on element matching given locator.

Tags:	action, mouse
Start / End / Elapsed:	20230405 17:25:21.693 / 20230405 17:25:21.698 / 00:00:00.005
17:25:21.694	INFO	Getting element with locator: WindowsElement(item=<uiautomation.uiautomation.TreeItemControl object at 0x00000262E60FBCA0>, locator=None, name='SEMM03', automation_id='', control_type='TreeItemControl', class_name='', left=1151, right=1201, top=280, bottom=298, width=50, height=18, xcenter=1176, ycenter=289)	
17:25:21.695	INFO	Returning element: WindowsElement(item=<uiautomation.uiautomation.TreeItemControl object at 0x00000262E60FBCA0>, locator=None, name='SEMM03', automation_id='', control_type='TreeItemControl', class_name='', left=1151, right=1201, top=280, bottom=298, width=50, height=18, xcenter=1176, ycenter=289)	
17:25:21.697	INFO	2023-04-05 17:25:21.697 action.py[190] _ click_element -> Can not move cursor. TreeItemControl's 
                                                                                  ===================================
BoundingRectangle is (0,0,0,0)[0x0]. SearchProperties: {ControlType: TreeItemControl}	
00:00:01.916KEYWORD Output Tree Items ${Window}, ${Group}

What have I done wrong ?

**Comment by cmin764** [comment]
Hi there @hhsem and thx for being so eager to test our latest library releases. Also your feedback provided so far was very valuable for us in order to develop the latest enhancements.

This "Can not move cursor." reported error in the logs is not something under our control and it is a known limitation expected to happen when using previously retrieved Windows elements after disturbing the automation or interacting with your keyboard/mouse during automation.
Luckily, you can use string locators and control the main app window before clicking elements in order to solve this. And as an extra safety measure, you can use a clicking helper which will test against this problem like so:
```robotframework
Click Group If Visible
    [Arguments]    ${name}

    ${group} =    Get Element    ${DATABASE} > name:"${name}"
    # Checks if the element is not visible (not having dimensions), thus not clickable
    #  for some reason. (when disturbing the automation)
    IF    ${group.width} <= ${0} or ${group.height} <= ${0}
        Fail    Group ${name} not clickable!
    END

    RPA.Windows.Click    ${group}    wait_time=${0.1}
```

Also, I enhanced your robot example automating KeePass in such a way it will avoid such issues by both using this helper and string locators. So please give it a try and let us know if it acts differently, even when you disturb the automation on purpose.

- Robot: https://github.com/cmin764/robots/tree/main/keepass-win
- Your enhanced task: `List Groups`
- Our newly added more robust task: `Iterate Groups`

Good luck and please close this Issue if it solves the problem!

**Comment by cmin764** [comment]
Also replied on your Forum post [here](https://forum.robocorp.com/t/print-tree-with-return-structure-reports-an-error/2438/28?u=cosmin).

**Comment by cmin764** [comment]
This should be fixed now with `rpaframework==22.5.1`.

**Comment by hhsem** [comment]
Thank you for  help and Information

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: Cosmin Poieana ***@***.***>
Gesendet: Friday, April 7, 2023 1:14:11 PM
An: robocorp/rpaframework ***@***.***>
Cc: Semmelmayer, Herbert ***@***.***>; Mention ***@***.***>
Betreff: Re: [robocorp/rpaframework] RPA.WINDOWS Click not successful but no error reported again (Issue #913)


This should be fixed now with rpaframework==22.5.1.
